substrate/frame/support/src/traits/tokens/fungible/regular.rs

```diff
@@ -408,18 +414,18 @@ pub trait Balanced<AccountId>: Inspect<AccountId> + Unbalanced<AccountId> {
 	///
 	/// This is just the same as burning and issuing the same amount and has no effect on the
 	/// total issuance.
-	fn pair(amount: Self::Balance) -> (Debt<AccountId, Self>, Credit<AccountId, Self>) {
-		(Self::rescind(amount), Self::issue(amount))
+	fn pair(amount: Self::Balance) -> (Credit<AccountId, Self>, Debt<AccountId, Self>) {
+		(Self::issue(amount), Self::rescind(amount))
```

Comment:
you have the similar issue even when you do these operation in a different order, a case when the total issuance is so big that `amount` saturates it. and this probably a bit more likely to happen than what you described in PR.

regardless, I would also not change the API, regardless the order of the operations, you can keep the api.
also these kind of changes should be mirrored for fungibles traits.

Reply:
Fixed this, good spot thanks